apps/vanilla/src/pages/task2/ts/classes/turn-generator.ts

```diff
@@ -0,0 +1,41 @@
+import { DiceGenerator } from './dice-generator';
+import { Publisher } from './publisher';
+
+/** Generator that determines the player's turn order. */
+export class TurnGenerator extends Publisher<number, DiceGenerator> {
+
+	private _currentPlayerId: number;
+
+	private static _instance: TurnGenerator | null = null;
+
+	private constructor() {
+		super();
+		this._currentPlayerId = 0;
+	}
+
+	/** Determines the next player to roll the dice. */
+	public getNextTurn(): void {
```

Comment:
Prefix `get` always returns sth
```suggestion
	public generateNextTurn(): void {
```